File: 06-Functions/Factorial.java
import java.util.*;

public class Factorial {
    public static void fact(int n) {
        if (n < 0) {
            System.out.println("Invalid Number");
            return;
        }
        int f = 1;
        for (int i = n; i >= 1; i--) {
            f = f * i;
        }
        System.out.println("Factorial is: " + f);
    }

    public static void main(String args[]) {
        System.out.println("Enter the number whose factorial you want to know:");
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        fact(n);

    }
}

File: C++/Recursion/Factorial.cpp
#include <iostream>
using namespace std;

int factorial(int n)
{
    // base case
    if (n == 0)
    {
        return 1;
    }
    return n * factorial(n - 1);
}
int main()
{

    int n;
    cout << "Enter the number:" << endl;
    cin >> n;

    int ans = factorial(n);
    cout << "Factorial of the number is:" << ans << endl;
}
